plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/connection/ConnectionWizardPageModel.java

```diff
@@ -303,8 +303,10 @@ public IStatus getConnectionCreationError() {
 	public IStatus connect() {
 		IStatus status = Status.OK_STATUS;
 		try {
-			if(connection != null) {
-				connection.connect();
+			if(connection != null && !connection.connect()) {
```

Comment:
@adietish This change was made to make sure to get the error and NOT add the connection to the explorer, but I also added some cases which will produce an exception that is caught by a subsequent throw.  The boolean return may not be needed here but probably requires us to discuss further.  We may even wish to remove the boolean return from the method signature or maybe identify it as something that indicates maybe that we are already connected?